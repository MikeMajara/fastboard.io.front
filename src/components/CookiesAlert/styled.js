import styled, { css, up } from '@xstyled/styled-components';
import { Link } from 'react-router-dom';

export const SWrapper = styled.div`
  position: fixed;
  bottom: 0;
  display: flex;
  width: 100%;
  height: 160px;
  box-sizing: border-box;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  padding-right: default;
  padding-left: default;
  background-color: cookiesAlertBg;
  font-size: sm;

  ${up('sm', css`
    height: 100px;
    flex-direction: row;
    justify-content: space-between;
  `)}

  ${up('md', css`
    height: 60px;
    flex-direction: row;
    justify-content: space-between;
  `)}
`;

export const SLink = styled(Link)`
  color: fg;
  text-decoration: underline;
`;
