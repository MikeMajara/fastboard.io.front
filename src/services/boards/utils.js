/**
 * Send boards event trough the network.
 *
 * @param {String} eventName Event key name.
 * @param {Object} data Data associated to the event.
 * @param {Function} ask Callback for server event acknowledgement.
 */
function send(eventName, data, ack) {
  this.socket.emit(eventName, data, ack);
}

/**
 * Check if the service is initialized.
 *
 * @return {Boolean} True if the service is initialized; false otherwise.
 */
function isInit() {
  return this.isInit;
}

export {
  isInit,
  send,
};
