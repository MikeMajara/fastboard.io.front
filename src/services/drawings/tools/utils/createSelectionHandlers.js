import { Group, Shape } from 'paper';

import { Log } from '#utils';
import { cursorTypes, selectionColorCode } from '#constants';
import store, { setSelectorCursorHover } from '#store';

import { bounds } from '../selector/constants';

const boundsArray = Object.values(bounds);
const bounds2cursorType = Object.freeze({
  [bounds.topLeft]     : cursorTypes.rotating,
  [bounds.topRight]    : cursorTypes.menu,
  [bounds.bottomLeft]  : cursorTypes.resizingSW,
  [bounds.bottomRight] : cursorTypes.resizingSE,
});

/**
 * Add selection handlers to the selected item.
 *
 * @param {Object} selectedItem Item to be added selection handlers to.
 * @param {Object} selectionLayer Layer that will contain the selection handlers.
 *
 * @returns {Object} Item - group with 4 handler items - from the selection handler layer.
 */
export default function createSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : createSelectionHandlers', { selectedItem });

  // Set selected item to show its selection state
  selectedItem.set({
    selected      : true,
    selectedColor : selectionColorCode,
    onMouseEnter  : () => store.dispatch(setSelectorCursorHover(cursorTypes.dragging)),
    onMouseLeave  : () => store.dispatch(setSelectorCursorHover()),
  });

  // Create bound handlers on the selection layer
  boundsArray.forEach((bound) => {
    selectionLayer.addChild(new Shape.Circle({
      name         : bound,
      center       : selectedItem.internalBounds[bound],
      radius       : 6,
      fillColor    : selectionColorCode,
      onMouseEnter : () => store.dispatch(setSelectorCursorHover(bounds2cursorType[bound])),
      onMouseLeave : () => store.dispatch(setSelectorCursorHover()),
    }));
  });

  // Create a group with the handlers to manipulate all of them at once
  selectionLayer.addChild(
    new Group({
      name     : 'selectionHandlers', // @todo to constants
      children : boundsArray.map(
        (bound) => selectionLayer.children[bound],
      ),
    }),
  );

  // Reproduce the selected item transformations on the handler items group
  selectionLayer.children.selectionHandlers.transform(selectedItem.matrix);

  // Except the scaling of the handler items themselves, which retain the original size
  selectionLayer.children.selectionHandlers.children.forEach((handlerItem) => {
    handlerItem.set({ scaling : 1 });
  });

  return selectionLayer.children.selectionHandlers;
}
