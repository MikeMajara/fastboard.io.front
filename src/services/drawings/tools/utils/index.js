export { default as createSelectionHandlers } from './createSelectionHandlers';
export { default as removeSelectionHandlers } from './removeSelectionHandlers';
export { default as resizeSelectionHandlers } from './resizeSelectionHandlers';
export { default as rotateSelectionHandlers } from './rotateSelectionHandlers';
export { default as translateSelectionHandlers } from './translateSelectionHandlers';
