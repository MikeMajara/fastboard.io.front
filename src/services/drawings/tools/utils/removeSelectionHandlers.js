import { Log } from '#utils';

/**
 * Remove selection handlers.
 *
 * @param {Object} selectedItem Item to be removed selection handlers from.
 * @param {Object} selectionLayer Layer that contains the selection handlers.
 */
export default function removeSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : removeSelectionHandlers');

  selectedItem.set({ selected : false });
  selectionLayer.removeChildren();
}
