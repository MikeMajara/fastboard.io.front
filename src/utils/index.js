export { default as Log } from './logger';
export { default as generateId } from './generateId';
export { default as point2net } from './point2net';
export { default as setPreventUnload } from './setPreventUnload';
export { default as throttle } from './throttle';
export { default as timeoutPromise } from './timeoutPromise';
export { default as triggerDownload } from './triggerDownload';
export { default as triggerUpload } from './triggerUpload';
export { default as validBoardId } from './validBoardId';
