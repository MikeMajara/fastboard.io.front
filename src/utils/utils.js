/**
 * Throttles a function call.
 *
 * @param {Function} callback Function to be throttled.
 * @param {Integer} delay Throttle delay in milliseconds.
 *
 * @return {Function} Function that throttles the execution of the callback.
 */
export const throttle = (callback, delay = 0) => {
  let previousTime = new Date().getTime();

  return function throttledCallback(args) {
    const currentTime = new Date().getTime();

    if ((currentTime - previousTime) >= delay) {
      previousTime = currentTime;
      callback(...args);
    }
  };
};

/**
 * Creates a Promise with a rejection timeout.
 *
 * @param {Function} callback Function for the promise. Receives res and rej functions.
 * @param {Integer} timeout Timeout for the promise to be rejected in milliseconds.
 *
 * @return {Promise} Promise with rejection timeout.
 */
export const timeoutPromise = (callback, timeout) => new Promise(
  (res, rej) => {
    const timeoutId = setTimeout(() => rej(new Error()), timeout);

    callback(
      (...args) => {
        clearTimeout(timeoutId);
        res(...args);
      },
      (...args) => {
        clearTimeout(timeoutId);
        rej(new Error(...args));
      },
    );
  },
);
