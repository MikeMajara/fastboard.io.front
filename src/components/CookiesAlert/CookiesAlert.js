import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { useTranslation, Trans } from 'react-i18next';

import routes from '#routes';

import Button from '../Button';

import { SWrapper, SLink } from './styled';

const CookiesAlert = ({ onAccept }) => {
  const { t } = useTranslation();

  const LinkComponent = useMemo(
    () => <SLink to={routes.cookies} />,
    [],
  );

  return (
    <SWrapper>
      <span>
        <Trans
          i18nKey="cookies.message"
          components={{ linkToLegal : LinkComponent }}
        />
      </span>
      <Button onClick={onAccept}>
        {t('cookies.accept')}
      </Button>
    </SWrapper>
  );
};

CookiesAlert.propTypes = {
  onAccept : PropTypes.func.isRequired,
};

export default CookiesAlert;
