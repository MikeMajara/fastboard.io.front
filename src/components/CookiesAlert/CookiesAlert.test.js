import React from 'react';
import { fireEvent } from '@testing-library/react';

import { render } from '#devTools/tests/utils';

import CookiesAlert from './CookiesAlert';

describe('Component : CookiesAlert', () => {
  test('Is visible and reacts to actions', () => {
    const onAcceptFn = jest.fn();
    const { getByText } = render(
      <CookiesAlert onAccept={onAcceptFn} />,
      { router : true },
    );

    expect(getByText('cookies.message')).toBeInTheDocument();
    fireEvent.click(getByText('cookies.accept'));
    expect(onAcceptFn).toHaveBeenCalledTimes(1);
  });

  test('Snapshot', () => {
    const { asFragment } = render(
      <CookiesAlert onAccept={jest.fn()} />,
      { router : true },
    );

    const componentRender = asFragment();
    expect(componentRender).toMatchSnapshot();
  });
});
