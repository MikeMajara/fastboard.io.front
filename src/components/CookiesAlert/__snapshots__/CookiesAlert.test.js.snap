// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : CookiesAlert Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-qZtCU lklQul"
  >
    <span>
      cookies.message
    </span>
    <button
      class="sc-qQYBZ bNCOou"
      type="primary"
    >
      cookies.accept
    </button>
  </div>
</DocumentFragment>
`;
